Sources/CoreMetrics/Metrics.swift

```diff
@@ -773,6 +942,42 @@ public protocol RecorderHandler: AnyObject, _SwiftMetricsSendableProtocol {
     func record(_ value: Double)
 }
 
+/// A `GaugeHandler` represents a backend implementation of a `Gauge`.
+///
+/// This type is an implementation detail and should not be used directly, unless implementing your own metrics backend.
+/// To use the SwiftMetrics API, please refer to the documentation of `Gauge`.
+///
+/// # Implementation requirements
+///
+/// To implement your own `GaugeHandler` you should respect a few requirements that are necessary so applications work
+/// as expected regardless of the selected `GaugeHandler` implementation.
+///
+/// - The `RecorderHandler` must be a `class`.
+public protocol GaugeHandler: AnyObject, _SwiftMetricsSendableProtocol {
```

Comment:
Should we align the name of this with the other name we choose?

Reply:
I would like to come back to Gauge after we do a deprecation cycle and have a major version, so would prefer to leave everything but the main type as Gauge